.github/workflows/agent-build-refactored.yml

```diff
@@ -123,10 +140,41 @@ jobs:
           image: tonistiigi/binfmt:qemu-v6.2.0
           platforms: all
 
+      - name: Install python
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Prepare environment.
+        uses: ./.github/actions/execute-runner
+        with:
+          runner-fqdn: "agent_build_refactored.__init__.BuildTestEnvironment"
+
+      - name: Prepare SSH
+        id: prepare-ssh
+        shell: bash
+        run: |
+          echo '${{ secrets.AWS_DEV_US_EAST_1_KEYPAIR_NAME }}' > /tmp/private_key.pem
+          chmod 600 /tmp/private_key.pem
+          eval `ssh-agent -s`
+          
+          cat "/tmp/private_key.pem" | ssh-add -
```

Comment:
I think these lines can be simplified (no need to write the key to the filesystem):
```
eval `ssh-agent -s`
echo '${{ secrets.AWS_DEV_US_EAST_1_KEYPAIR_NAME }}' | ssh-add -
```

Reply:
Hm maybe I'm wrong because I see the `aws_private_key_path: "/tmp/private_key.pem"` below

In that case why use both ssh-agent and store the key on the filesystem?